openedx_lti_tool_plugin/tests/test_views.py

```diff
@@ -312,48 +312,235 @@ def test_enroll_without_enrollment(self, course_enrollment_mock: MagicMock):
         return_value=MagicMock(status_code=200, content='random-content'),
     )
     @patch.object(LtiToolLaunchView, 'enroll', return_value=None)
+    @patch.object(CourseKey, 'from_string', return_value='random-course-key')
     @patch.object(LtiToolLaunchView, 'authenticate_and_login', return_value='random-user')
+    @patch.object(CourseAccessConfiguration.objects, 'filter')
+    @patch('openedx_lti_tool_plugin.views.COURSE_ACCESS_CONFIGURATION')
+    @patch.object(DjangoMessageLaunch, 'get_launch_data', return_value={**BASE_LAUNCH_DATA, 'azp': AUD})
+    @patch('openedx_lti_tool_plugin.views.DjangoCacheDataStorage')
+    @patch('openedx_lti_tool_plugin.views.DjangoDbToolConf')
+    @patch.object(DjangoMessageLaunch, '__init__', return_value=None)
+    def test_post_with_course_access_configuration_enabled(
+        self,
+        message_launch_mock: MagicMock,
+        tool_conf_mock: MagicMock,
+        tool_storage_mock: MagicMock,
+        get_launch_data_mock: MagicMock,
+        course_access_configuration_mock: MagicMock,
+        course_access_configuration_filter_mock: MagicMock,
+        authenticate_and_login_mock: MagicMock,
+        course_key_mock: MagicMock,
+        enroll_mock: MagicMock,
+        redirect_mock: MagicMock,
+    ):
+        """Test POST request with COURSE_ACCESS_CONFIGURATION flag enabled.
+
+        Args:
+            message_launch_mock: Mocked DjangoMessageLaunch class.
+            tool_conf_mock: Mocked DjangoDbToolConf class.
+            tool_storage_mock: Mocked DjangoCacheDataStorage class.
+            get_launch_data_mock: Mocked DjangoMessageLaunch get_launch_data method.
+            course_access_configuration_mock: Mocked COURSE_ACCESS_CONFIGURATION object.
+            course_access_configuration_filter_mock: Mocked CourseAccessConfiguration.ojects filter method.
+            authenticate_and_login_mock: Mocked LtiToolLaunchView authenticate_and_login method.
+            course_key_mock: Mocked CourseKey from_string method.
+            enroll_mock: Mocked enroll LtiToolLaunchView method.
+            redirect_mock: Mocked redirect function.
+        """
+        course_access_configuration_mock.is_enabled.return_value = True
```

Comment:
What do you think of using this decorator https://github.com/openedx/edx-toggles/blob/master/edx_toggles/toggles/testutils.py#L7 to mock the value of the waffle flag?